syntax = "proto3";

package devices;


message Inverter {
	optional double timestamp = 1;
	optional double temperature = 2;
	optional double motor_temp = 3;
	optional double torque = 4;
	optional double speed = 5;
}

message Gps {
	optional double timestamp = 1;
	optional string msg_type = 2;
	optional string time = 3;
	optional double latitude = 4;
	optional double longitude = 5;
	optional double altitude = 6;
	optional uint32 fix = 7;
	optional uint32 satellites = 8;
	optional string fix_state = 9;
	optional double age_of_correction = 10;
	optional double course_over_ground_degrees = 11;
	optional double course_over_ground_degrees_magnetic = 12;
	optional double speed_kmh = 13;
	optional string mode = 14;
	optional double position_diluition_precision = 15;
	optional double horizontal_diluition_precision = 16;
	optional double vertical_diluition_precision = 17;

	optional bool heading_valid = 18;
	optional double heading_vehicle = 19;
	optional double heading_motion = 20;
	optional double heading_accuracy_estimate = 21;
	optional double speed_accuracy = 22;
}



message GpsVec {
	repeated Gps gps = 1;
}

message InverterVec {
	repeated Inverter inverter = 1
}
